sympy/stats/tests/test_stochastic_process.py

```diff
@@ -117,6 +154,18 @@ def test_DiscreteMarkovChain():
     assert E(X[1]**2, Eq(X[0], 1)) == Rational(8, 3)
     assert variance(X[1], Eq(X[0], 1)) == Rational(8, 9)
     raises(ValueError, lambda: E(X[1], Eq(X[2], 1)))
+    raises(ValueError, lambda: DiscreteMarkovChain('X', [0, 1], T))
+
+    # testing miscellaneous queries with different state space
+    X = DiscreteMarkovChain('X', ['A', 'B', 'C'], T)
+    assert P(Eq(X[1], 2) & Eq(X[2], 1) & Eq(X[3], 0),
+    Eq(P(Eq(X[1], 0)), Rational(1, 4)) & Eq(P(Eq(X[1], 1)), Rational(1, 4))) == Rational(1, 12)
+    assert P(Eq(X[2], 1) | Eq(X[2], 2), Eq(X[1], 1)) == Rational(2, 3)
+    assert P(Eq(X[2], 1) & Eq(X[2], 2), Eq(X[1], 1)) is S.Zero
+    assert P(Ne(X[2], 2), Eq(X[1], 1)) == Rational(1, 3)
+    assert str(E(X[1] ** 2, Eq(X[0], 1))) == "A**2/3 + 2*C**2/3"
+    assert str(variance(X[1], Eq(X[0], 1))) == "2*(-A/3 + C/3)**2/3 + (2*A/3 - 2*C/3)**2/3"
```

Comment:
can this be equality be checked without casting to string?

Reply:
It was done a little further up in the tests before. The only other way I could think of is to do something like `free_symbols`.

Reply:
Why do you need `free_symbols`?

Reply:
The symbols `A` and `C` are not defined by the test class. But I realised now that I can use `state_space` to access them instead.

Reply:
Yes, thanks for removing `str( ... )`. Using `str` in the tests may break a lot of stuff if we decide to change the way SymPy expressions are rendered to strings.